### PR TITLE
Add various pool list APIs and correct db

### PR DESCRIPTION
**Pull request by Jaxonwht**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -26,11 +26,6 @@ repos:
         args:
           - --max-line-length=140
 
-  - repo: https://github.com/pycqa/isort
-    rev: 5.10.1
-    hooks:
-      - id: isort
-
```
Comment (jieqiu0630) [inline .pre-commit-config.yaml:33]: Can we keep this?

File: web/dal/pool_list_dal.py [hunk 2]
```diff
@@ -1,10 +1,44 @@
...
 from typing import List
 
 from init import db
-from models.user_data import PoolList
+from models.series_data import PoolInfo
+from models.user_data import NamedPoolList
 
 
-def add_pool_list_to_db(list_name: str, pool_list: List[str]):
+def create_or_append_pool_list(list_name: str, pool_list: List[str]):
```
Comment (jieqiu0630) [inline web/dal/pool_list_dal.py]: Let's add `-> None` here as well.

File: web/migrations/versions/b03b5f5fb4a4_.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        sa.ForeignKeyConstraint(
+            ["list_name"],
+            ["named_pool_list.list_name"],
+        ),
+        sa.ForeignKeyConstraint(
+            ["pool_info_name"],
+            ["pool_info.pool_name"],
+        ),
+        sa.PrimaryKeyConstraint("pool_info_name", "list_name"),
+    )
```
Comment (jieqiu0630) [inline web/migrations/versions/b03b5f5fb4a4_.py:34]: 😍

File: web/views/api.py [hunk 4]
```diff
@@ -2,21 +2,44 @@
...
 
 
-@app.route("/add-pool-list", methods=["POST"])
+@app.route("/add-pool-list", methods=["PUT", "PATCH", "POST"])
 def add_pool_list():
     """
     Add a list of your favorite pools to the `pool_list` database.
+
+    The PUT request body should contain a list_name, and a list of strings
+    in the parameter pool_list.
```
Comment (jieqiu0630) [inline web/views/api.py:21]: Nit: "the request body"

**Review by jieqiu0630 — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Jaxonwht) [reply .pre-commit-config.yaml:33]: Conflict with black hook

Comment (jieqiu0630) [reply .pre-commit-config.yaml:33]: Added it back with extra flags so that it's now compliant with the black hook.

**Pull request merged**